stock_barcodes_gs1/wizard/stock_barcodes_read.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2019 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, models
+
+
+class WizStockBarcodesRead(models.AbstractModel):
+    _inherit = "wiz.stock.barcodes.read"
+
+    def _prepare_lot_values(self, barcode_decoded):
+        lot_barcode = barcode_decoded.get("10", False)
+        return {
+            "name": lot_barcode,
+            "product_id": self.product_id.id,
+            "company_id": self.env.user.company_id.id,
+        }
+
+    def _create_lot(self, barcode_decoded):
+        return self.env["stock.production.lot"].create(
+            self._prepare_lot_values(barcode_decoded)
+        )
+
+    def process_lot(self, barcode_decoded):
+        lot_barcode = barcode_decoded.get("10", False)
+        lot = self.env["stock.production.lot"].search(
+            [("name", "=", lot_barcode), ("product_id", "=", self.product_id.id)]
+        )
+        if not lot:
+            lot = self._create_lot(barcode_decoded)
+        self.lot_id = lot
+
+    def process_barcode(self, barcode):
+        """Only has been implemented AI (01, 02, 10, 37), so is possible that
+        scanner reads a barcode ok but this one is not precessed.
+        """
+        try:
+            barcode_decoded = self.env["gs1_barcode"].decode(barcode)
+        except Exception:
```

Comment:
Is that generic catch wanted ?

Reply:
Well, checking the code it can be replace by ValueError and ValidationError, however, I think it is right to keep it generic. First we try to transform to GS1, if it works we proceed, if it fails, we want to use the standard behavior.

Reply:
Well, checking the code it can be replace by ValueError and ValidationError, however, I think it is right to keep it generic. First we try to transform to GS1, if it works we proceed, if it fails, we want to use the standard behavior.